contracts/compliance/modular/modules/InitialLockupPeriodModule.sol

```diff
@@ -0,0 +1,160 @@
+//                                          +@@@@@. =@@@=  .+@@@@@*:
+//                                          #@@@@#:%@@#. :*@@@@#-
+//                                          @@@@@%@@@= :#@@@@+.
+//                                         :@@@@@@@#.:#@@@%-
+//                                         +@@@@@@-.*@@@*:
+//                                         #@@@@#.=@@@+.
+//                                         @@@@+-%@%=
+//                                        :@@@#%@%=
+//                                        +@@@@%-
+//                                        :#%%=
+//
+/**
+ *     NOTICE
+ *
+ *     The T-REX software is licensed under a proprietary license or the GPL v.3.
+ *     If you choose to receive it under the GPL v.3 license, the following applies:
+ *     T-REX is a suite of smart contracts implementing the ERC-3643 standard and
+ *     developed by Tokeny to manage and transfer financial assets on EVM blockchains
+ *
+ *     Copyright (C) 2024, Tokeny sàrl.
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *     This specific smart contract is also licensed under the Creative Commons
+ *     Attribution-NonCommercial 4.0 International License (CC-BY-NC-4.0),
+ *     which prohibits commercial use. For commercial inquiries, please contact
+ *     Tokeny sàrl for licensing options.
+ */
+
+pragma solidity 0.8.27;
+
+import "./AbstractModuleUpgradeable.sol";
+
+error InsufficientBalanceTokensLocked(address user, uint256 value, uint256 availableAmount);
+
+event LockupPeriodSet(address indexed compliance, uint256 lockupPeriod);
+
+
+contract InitialLockupPeriodModule is AbstractModuleUpgradeable {
+
+    struct LockedTokens {
+        uint256 amount;
+        uint256 releaseTimestamp;
+    }
+
+    mapping(address compliance => uint256 lockupPeriod) private _lockupPeriods;
+    mapping(address compliance => mapping(address user => LockedTokens[] lockedTokens)) private _lockedTokens;
+
+    /// @dev initializes the contract and sets the initial state.
+    function initialize() external initializer {
+        __AbstractModule_init();
+    }
+
+    /// @dev sets the lockup period for a compliance contract.
+    /// @param _lockupPeriod the lockup period in seconds.
+    function setLockupPeriod(uint256 _lockupPeriod) external onlyComplianceCall {
```

Comment:
probably more convenient for users to set a number of days, and the smart contract should transfor that in seconds as UNIX timestamp is calculated in seconds